src/heaps/mutable_binary_heap.jl

```diff
@@ -101,23 +101,23 @@ function _heap_bubble_down!(comp::Comp,
 end
 
 function _binary_heap_pop!(comp::Comp,
-    nodes::Vector{MutableBinaryHeapNode{T}}, nodemap::Vector{Int}) where {Comp,T}
+    nodes::Vector{MutableBinaryHeapNode{T}}, nodemap::Vector{Int}, nd_id::Int) where {Comp,T}
 
-    # extract root node
-    rt = nodes[1]
+    # extract  node
+    rt = nodes[nd_id]
     v = rt.value
     @inbounds nodemap[rt.handle] = 0
 
     if length(nodes) == 1
         # clear
         empty!(nodes)
     else
-        # place last node to root
-        @inbounds nodes[1] = new_rt = pop!(nodes)
-        @inbounds nodemap[new_rt.handle] = 1
+        # place last node to removed one
+        @inbounds nodes[nd_id] = new_rt = pop!(nodes)
+        @inbounds nodemap[new_rt.handle] = nd_id
 
         if length(nodes) > 1
-            _heap_bubble_down!(comp, nodes, nodemap, 1)
+            _heap_bubble_down!(comp, nodes, nodemap, nd_id)
```

Comment:
Since we're not always replacing the root node any more, we'll need to call `_heap_bubble_up` here if the new node value is less than its parent (or alternatively, less than the node it replaces, since that's easier to determine).

Reply:
Should also add a test for this situation.

Reply:
Ya! Sure! Sorry missed it.